verifier/src/qe_identity.rs

```diff
@@ -136,21 +136,22 @@ impl QeIdentity {
         sgx_attributes_t { flags, xfrm }.into()
     }
 
-    fn verify(&self, time: DateTime) -> Result<(), Error> {
+    fn verify(&self, time: Option<DateTime>) -> Result<(), Error> {
         self.verify_version()?.verify_time(time)?;
         Ok(())
     }
 
-    fn verify_time(&self, time: DateTime) -> Result<&Self, Error> {
+    fn verify_time(&self, time: Option<DateTime>) -> Result<&Self, Error> {
         let issue_date = self.issue_date.parse::<DateTime>()?;
         let next_update = self.next_update.parse::<DateTime>()?;
-        if time < issue_date {
-            Err(Error::QeIdentityNotYetValid)
-        } else if time >= next_update {
-            Err(Error::QeIdentityExpired)
-        } else {
-            Ok(self)
+        if let Some(date_time) = time {
+            if date_time < issue_date {
+                return Err(Error::QeIdentityNotYetValid);
+            } else if date_time >= next_update {
+                return Err(Error::QeIdentityExpired);
+            }
         }
+        Ok(self)
```

Comment:
Not sure it's better, I could see arguments for concern on that final `Some(_)`
```suggestion
        match time {
            Some(date_time) if date_time < issue_date => Err(Error::QeIdentityNotYetValid),
            Some(date_time) if date_time >= next_update => Err(Error::QeIdentityExpired),
            None | Some(_) => Ok(self),
        }
```

Reply:
Yeah, I was wondering what it would look like to do this with pattern matching. I don't know if I like this. In my opinion this is a case of just because we can use a feature it doesn't mean we should. I think the original is more readable, but I could be convinced otherwise.

Reply:
sounds good, I don't have a strong preference either way, the match guards can be a little dense to read at times.